### PR TITLE
fix pci watchdog plug into pcie.0 issue

### PR DESCRIPTION
**Pull request by xutian**
ID: 1809396

Signed-off-by: Xu Tian <xutian@redhat.com>

### CODE REVIEW

**Comment by xutian** [comment]
Deps on https://github.com/avocado-framework/avocado-vt/pull/2457

**Comment by yanan-fu** [comment]
@xutian  I am not confident about watchdog device, no pcie for watchdog ?

File: qemu/tests/watchdog.py [hunk 1]
```diff
@@ -283,8 +284,14 @@ def hotplug_unplug_watchdog_device():
                 test.fail("Can find watchdog pci")
 
         plug_watchdog_device = params.get("plug_watchdog_device", "i6300esb")
+        machine_type = params.get("machine_type")
         watchdog_device_add = ("device_add driver=%s, id=%s"
                                % (plug_watchdog_device, "watchdog"))
+        if plug_watchdog_device == "i6300esb" and machine_type == "q35":
```
Comment (yanan-fu) [inline qemu/tests/watchdog.py]: `test_type = hotplug_unplug_watchdog_device` is for `i6300esb` only. May be we can use `pcie-pci-bridge-0` directly ?

Comment (xutian) [reply qemu/tests/watchdog.py]: I'm not sure that old version's qemu support 'pcie-pci-bridge' too.

Comment (jingzhao84) [reply qemu/tests/watchdog.py]: @yanan-fu @xutian  I had checked RHEL7.8 qemu version and there have pcie-pci-bridge through qemu-kvm -device ?

File: qemu/tests/watchdog.py [hunk 2]
```diff
@@ -283,8 +284,14 @@ def hotplug_unplug_watchdog_device():
...
 
         plug_watchdog_device = params.get("plug_watchdog_device", "i6300esb")
+        machine_type = params.get("machine_type")
         watchdog_device_add = ("device_add driver=%s, id=%s"
                                % (plug_watchdog_device, "watchdog"))
+        if machine_type == "q35":
+            bus = "pcie-pci-bridge-0"
+            if not vm.devices.has_device('pcie-pci-bridge'):
+                bus = "pci-bridge-0"
+            watchdog_device_add += ",bus=%s,addr=0x1f" % bus
```
Comment (yanan-fu) [inline qemu/tests/watchdog.py]: Only one watchdog hotplug case, no need to add common function in framework. And `addr=0x1f` is safe.

Acked-by: Yanan Fu <yfu@redhat.com>

File: qemu/tests/watchdog.py (hunk 1 above)
Comment (xutian) [reply qemu/tests/watchdog.py]: Ok, let hard code bus to pcie-pci-bridge-0.

**Comment by xutian** [comment]
@yanan-fu  and @jingzhao84  commit update, would you like to review again. 

Thanks,
Xu

**Pull request merged**